src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2419,12 +2419,6 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
             assert(simdSize != 64 || IsBaselineVector512IsaSupportedDebugOnly());
 
-            if ((simdBaseType == TYP_BYTE) || (simdBaseType == TYP_UBYTE))
```

Comment:
Wonder if it makes sense to check if IsBaselineVector512IsaSupportedOpportunistically() here instead of in gentree,cpp for non simd64

```
if (((simdBaseType == TYP_BYTE) || (simdBaseType == TYP_UBYTE)) && !IsBaselineVector512IsaSupportedDebugOnly())
{
               // byte/sbyte multiplication reuires Avx512 support
                break;
}
```

Reply:
Wouldn't this check prevent the gentree.cpp to be called unless AVX512F is supported?
Current implementation can take advantage of AVX512 for simd32/simd16 or fallback to AVX2/SSE41 depending on available support.

Reply:
My apologies if I misunderstood. I assumed [this](https://github.com/dotnet/runtime/pull/86811/files#diff-fba867eda1c745875748370c59da2195c3823b85a17d199444068d93f91c749bR20195) means that for byte types and non-simd64 types, we go to software fallback anyway. Feel free to correct me if I got that wrong

Reply:
If that check fails, it will fall-through to [this](https://github.com/dotnet/runtime/pull/86811/files#diff-fba867eda1c745875748370c59da2195c3823b85a17d199444068d93f91c749bR20296), splitting each simd16/simd32/simd64 input into a pair of hi-lo simd16/simd32/simd64. Multiplication is then performed on both halves, and hi-lo results are then narrowed and merged together.
Basically, the if/else are attempts to use intrinsics that avoid the split/merge, using the next larger simd available. If it's not possible, then the fallback approch is used (which is valid whatever the simd size).

Reply:
Thanks for the explanation. Makes sense now